fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -1177,6 +1177,29 @@ public void testNoUnion() {
         }
     }
 
+    @Test
+    public void testDryRunSaveRecord() throws Exception {
```

Comment:
It would be nice to have more tests here so that we're more confident we're not accidentally update the database in dry-run mode. One here that seems like we'd want to check is indexes: if you save a record with an indexed field in dry-run mode, then you shouldn't find index entries from that record when scanning the index.